BSD builds are broken

```
p/io_ptrace.c:193:25: error: use of undeclared identifier 'PTRACE_GETSIGINFO'
        ret = r_io_ptrace (io, PTRACE_GETSIGINFO, pid, NULL, &sig);
                               ^
1 error generated.
```

thisis a linuxism, can you add a #if __linux__ ?

> On 17 Oct 2019, at 12:22, Anton Kochkov <notifications@github.com> wrote:
> 
> p/io_ptrace.c:193:25: error: use of undeclared identifier 'PTRACE_GETSIGINFO'
>         ret = r_io_ptrace (io, PTRACE_GETSIGINFO, pid, NULL, &sig);
>                                ^
> 1 error generated.
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub <https://github.com/radareorg/radare2/issues/15296?email_source=notifications&email_token=AAG75FRS2J7B7YB3MTGRMU3QPA4FTA5CNFSM4JBXM662YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4HSNIMLA>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAG75FR4O7K4AQONVTTYTMLQPA4FTANCNFSM4JBXM66Q>.
>

--- a/libr/io/p/io_ptrace.c
+++ b/libr/io/p/io_ptrace.c
@@ -227,24 +227,24 @@ static RIODesc *__open(RIO *io, const char *file, int rw, int mode) {
 				break;
 			}
 #endif
+			return NULL;
 		} else if (__waitpid (pid)) {
 			ret = pid;
 		} else {
 			eprintf ("Error in waitpid\n");
-		}
-	}
-
-	if (ret != -1) {
-		RIOPtrace *riop = R_NEW0 (RIOPtrace);
-		if (!riop) {
 			return NULL;
 		}
-
-		riop->pid = riop->tid = pid;
-		open_pidmem (riop);
-		desc = r_io_desc_new (io, &r_io_plugin_ptrace, file, rw | R_PERM_X, mode, riop);
-		desc->name = r_sys_pid_to_path (pid);
 	}
+
+	RIOPtrace *riop = R_NEW0 (RIOPtrace);
+	if (!riop) {
+		return NULL;
+	}
+
+	riop->pid = riop->tid = pid;
+	open_pidmem (riop);
+	desc = r_io_desc_new (io, &r_io_plugin_ptrace, file, rw | R_PERM_X, mode, riop);
+	desc->name = r_sys_pid_to_path (pid);
 
 	return desc;
 }

